### PR TITLE
Floating App Bar Recipe

### PR DESCRIPTION
**Pull request by brianegan**
A recipe describing how to use the CustomScrollView and SliverAppBar together to create a floating app bar!

### CODE REVIEW

File: src/docs/cookbook/lists/floating-app-bar.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+  title: Working with long lists
+  path: /docs/cookbook/lists/long-lists
+---
+
+To make it easier for users to view a list of items, you may want to hide the
+app bar as the user scrolls down the list. This is especially true if your app
+displays a "tall" app bar that occupies a lot of vertical space.
+
+Traditionally, you create an app bar by providing an `appBar` property to the
+`Scaffold` Widget. This will create a fixed app bar that always remains above
```
Comment (sfshaza2) [inline src/docs/cookbook/lists/floating-app-bar.md]: will create => creates

File: src/docs/cookbook/lists/floating-app-bar.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+the `body` of the `Scaffold`.
+
+If you'd like to create a floating app bar scrolls offscreen as the user scrolls
```
Comment (sfshaza2) [inline src/docs/cookbook/lists/floating-app-bar.md]: floating app bar scrolls => floating app bar that scrolls .  ??

File: src/docs/cookbook/lists/floating-app-bar.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+through a list, you need to move the app bar from the Scaffold into a
+[`CustomScrollView`](https://docs.flutter.io/flutter/widgets/CustomScrollView-class.html).
+This allows you to create an app bar that scrolls offscreen as you scroll
+through a list of items contained inside the `CustomScrollView`!
```
Comment (sfshaza2) [inline src/docs/cookbook/lists/floating-app-bar.md]: This feels a little repetitive with the first sentence. How about:

Moving the app bar from the Scaffold into the [`CustomScrollView`][] allows you to create an
app bar that scrolls offscreen as you scroll through a list of items contained inside the
`CustomScrollView`.

File: src/docs/cookbook/lists/floating-app-bar.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+
+In order to create a floating app bar, you need to place the app bar inside a
+`CustomScrollView` that also contains the list of items. This synchronizes the
+scroll position of the app bar and the list of items. You may think of the
+`CustomScrollView` Widget as a `ListView` that allows you to mix and match
+different types of scrollable lists and widgets together!
+
+The scrollable lists and widgets that can be provided to the the
+`CustomScrollView` are known as slivers. There are several types of Slivers,
+such as a `SliverList`, `SliverGridList`, and `SliverAppBar`! In fact, the
```
Comment (sfshaza2) [inline src/docs/cookbook/lists/floating-app-bar.md]: Global. Replace the exclamation points with periods. :)

File: src/docs/cookbook/lists/floating-app-bar.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+    ),
+  ],
+);
+```
+
+{{site.alert.tip}}
+Play around with the [various properties you can pass to the `SliverAppBar`
+Widget](https://docs.flutter.io/flutter/material/SliverAppBar/SliverAppBar.html)
+and use hot reload to see the results. For example, you can use an `Image`
+Widget for the `flexibleSpace` to create a background image that shrinks in size
```
Comment (sfshaza2) [inline src/docs/cookbook/lists/floating-app-bar.md]: for the `flexibleSpacd` to => for the `flexibleSpace` property to

File: src/docs/cookbook/lists/floating-app-bar.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+`CustomScrollView`! You have two options: a
+[`SliverList`](https://docs.flutter.io/flutter/widgets/SliverList-class.html) or
+a [`SliverGrid`](https://docs.flutter.io/flutter/widgets/SliverGrid-class.html).
+If you need to display a list of items one after the other, use the `SliverList`
+Widget. If you need to display a grid list, use the `SliverGrid` Widget!
+
+The `SliverList` and `SliverGrid` Widgets take one required parameter: a
+[`SliverChildDelegate`](https://docs.flutter.io/flutter/widgets/SliverChildDelegate-class.html).
+While this sounds fancy, the delegate is simply used to provide a list Widgets
+to to `SliverList` or `SliverGrid`! For example, the
```
Comment (sfshaza2) [inline src/docs/cookbook/lists/floating-app-bar.md]: to to => to

File: src/docs/cookbook/lists/floating-app-bar.md [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+[`SliverChildBuilderDelegate`](https://docs.flutter.io/flutter/widgets/SliverChildBuilderDelegate-class.html)
+allows you to create a list of items that are built lazily as your scroll, just
```
Comment (sfshaza2) [inline src/docs/cookbook/lists/floating-app-bar.md]: your scroll => you scroll

File: src/docs/cookbook/lists/floating-app-bar.md (hunk 1 above)
Comment (brianegan) [reply src/docs/cookbook/lists/floating-app-bar.md]: Ah, I just can't stop myself from making this mistake, haha :)

**Comment by brianegan** [comment]
Thanks for the review, @Sfshaza :) I've addressed all feedback. Please let me know if it looks good or if ya see anything else that needs improvement.

**Pull request merged**